### PR TITLE
Reset user password

### PR DESCRIPTION
**Pull request by timwangmusic**
## Description
The user password reset process is as follows:
* User clicks on “forgot password” on login page
* Redirect them to recover password page to enter emails
* Backend sends an email for resetting password
* User clicks on the included link and fill out a form with new password

## Solution
* Add forgot password page and reset password page
* Enhance mailer to send password reset emails

## Testing
- [x] Integration testing on Heroku testing environment
- [x] Added new unit tests

## Checks
- [x] Have you removed commented code?
- [x] Have you used gofmt to format your code?

### CODE REVIEW

**Review by rwangsc18 — COMMENTED** [review]
Very nice changes. Great work!
I was testing it in the testing app, everything works such as reset link to email etc, except my new password didn't get updated in the system. Is it because of the environment?

**Comment by rwangsc18** [comment]
> Very nice changes. Great work! I was testing it in the testing app, everything works such as reset link to email etc, except my new password didn't get updated in the system. Is it because of the environment?

Just tried in the testing app. The password works as expected now.

File: templates/login_page.html [hunk 1]
```diff
@@ -67,6 +67,9 @@ <h1> Vacation Planner </h1>
             <button type="submit" class="btn btn-primary login-btn btn-block">Sign in</button>
             <!-- TODO: Add Login buttons for Social media (FB, G, TWT) as a separate PR -->
         </form>
+        <div class="mt-1">
+            <p><a href="/v1/forgot-password">forgot password?</a></p>
```
Comment (rwangsc18) [inline templates/login_page.html:71]: maybe we can make "forgot-password" also a button. For example, we can use "Login with Google" as a reference?

File: assets/js/reset-password.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    const oldPassword = document.querySelector('#old-password').value;
+    const newPassword = document.querySelector('#new-password').value;
+    const backendUrl = '/v1/reset-password-backend'
+    const data = {
+        'email': email,
+        'old_password': oldPassword,
+        'new_password': newPassword,
+    }
+    console.log(data);
+    await axios.put(
```
Comment (rwangsc18) [inline assets/js/reset-password.js:26]: in a separate PR, we can consider adding some UI to inform users about the successful password update. 

Currently after the password update, the users will stay on the reset password page. Maybe we can also redirect the to home search page or the login page, after users updated the password?

**Review by rwangsc18 — APPROVED** [review]
LGTM, approval. Left a few comments, we can consider in future PRs.

File: assets/js/reset-password.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+$('#password-reset-submit-btn').click(() => $('#reset-email-sent-alert').removeClass('d-none'));
+
```
Comment (sananand007) [inline assets/js/reset-password.js:2]: are these local constants ?

**Review by sananand007 — APPROVED** [review]
lg .. nice work 🌮

File: templates/login_page.html (hunk 1 above)
Comment (timwangmusic) [reply templates/login_page.html:71]: "forgot-password" is typically a link on most websites, I am adjusting the link to align with the sign in button.

File: assets/js/reset-password.js (hunk 2 above)
Comment (timwangmusic) [reply assets/js/reset-password.js]: Done.

File: assets/js/reset-password.js (hunk 3 above)
Comment (timwangmusic) [reply assets/js/reset-password.js:2]: Staled.

**Pull request merged**